src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThroughFallThroughLotsOfComment.java

```diff
@@ -0,0 +1,61 @@
+/*
+FallThrough
+checkLastCaseGroup = (default)false
+reliefPattern = (default)falls?[ -]?thr(u|ough)
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.fallthrough;
+
+public class InputFallThroughFallThroughLotsOfComment {
+
+    void method(int lineNumber) {
+        switch (lineNumber) {
+            case 3:
+                Object pi = null;
+                int line = 2;
+                // fall-through intended - the line contains already the first entries
+                // the following 4 lines contain additional satellite ids
+            case 4:
```

Comment:
There is no violation, which I am fine with since there is the words "fall-through" to document it is intended.
However, documentation specifically says `must be on the last non-empty line before the case triggering the warning` which is not the case here. Technically the last non-empty line before the case triggering the warning is `the following 4 lines contain additional satellite ids` which does not contain the key words.

Can you confirm behavior being seen is intended and documentation is slightly misleading. If so, documentation needs to be updated.
@romani

Reply:
@rnveach @romani should we have to strictly follow that `// fall-through` must be on the previous line before the case and no other text than fall-through should be there to suppress the violation?

Reply:
If something is not clear, follow documentation and old version of code. If documentation is unclear or feels conflicting, then ask here about the unclearness and conflict.

This issue is not about changing current behavior unless it is unavoidable.

> follow that // fall-through must be on the previous line before the case 

This is what documentation says. Let me know what isn't clear.

> no other text than fall-through should be there 

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java#L526
Old code is just a `find`. So there can be other text with the key word, but it needs to be the complete key word in one comment.

Reply:
I think it is reasonable, imagine big block(two screens) of case and comment of top. After you scroll, at the end of case you will forget about FallThrough comment. Placing such comment at the end is like INSTEAD of `break`.

Reply:
Here is example that fallthough comment is not absolutely last line, but still reasonable
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/594a1d9_2023221254/reports/diff/Orekit/index.html#A1

@rnveach , I think we need to consider to change documentation. Do you agree ?

Reply:
> consider to change documentation

Documentation says it must be last line and code followed that, which means that is the current functionality. What you are suggesting is a functionality change, not a documentation change.

````
$ cat TestClass.java
public class TestClass {
    void method() {
switch (i) {
  case 1:
    i++;
    // fall through
    // random
  case 2: // OK
    i++;
    // random
    // fallthru
  case 3: { // OK
    i++;
  }
}
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>

    <module name="TreeWalker">
<module name="FallThrough"/>
    </module>
</module>

$ java -jar checkstyle-10.10.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[WARN] TestClass.java:8:3: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
````

https://github.com/checkstyle/checkstyle/pull/12966#discussion_r1205417984
Says why I thought it was ok at first.
https://github.com/checkstyle/checkstyle/pull/12966#discussion_r1205448883
Says why I rethought about it and it might not be a good change.

We will need a new issue if you want to make the change.

Reply:
yes, we can consider this change as mostly ok and update doc reflect that we allow special comment word in last block of comment lines.
OR
we should keep violation if last comment is not fallthrough.

@rnveach, How user can fix this code to allow  fallthrough and in same time keep comments over/for case blocks
```
  // line 3 contains the number of satellites
  case 3:
    pi.maxSatellites = Integer.parseInt(line.substring(4, 6).trim());
    // fall-through intended - the line contains already the first entri212 
  // the following 4 lines contain additional satellite ids
  case 4:
  case 5:
  case 6:
  case 2: {
  
  }
```

@Kevin222004, please add this snippet to Inputs.

Reply:
I see rule like:
consider comments from next case up-to/back-to empty line or any code.

```
  // line 3 contains the number of satellites
  case 3:
    pi.maxSatellites = Integer.parseInt(line.substring(4, 6).trim());
    // fall-through intended - the line contains already the first entri212 
    // the following 4 lines contain additional satellite ids
    // some text
  case 4:
```

violation:
```
  // line 3 contains the number of satellites
  case 3:
    pi.maxSatellites = Integer.parseInt(line.substring(4, 6).trim());
    // fall-through intended - the line contains already the first entri212 
    // the following 4 lines contain additional satellite ids
    
    // some text
  case 4:
```
violation:
```
  // line 3 contains the number of satellites
  case 3:
    pi.maxSatellites = Integer.parseInt(line.substring(4, 6).trim());
    // fall-through intended - the line contains already the first entri212 
    // some text 
    
    // the following 4 lines contain additional satellite ids
  case 4:
```

Reply:
> We will need a new issue if you want to make the change.

@rnveach yes it would be good if we open a new issue for it or to implement rule like https://github.com/checkstyle/checkstyle/pull/12966#discussion_r1209384049

Reply:
> @rnveach, How user can fix this code to allow fallthrough and in same time keep comments over/for case blocks

You can't without changing something, either comment placement, fall through, or the check itself. If you want to change behavior, you need to open a new issue.